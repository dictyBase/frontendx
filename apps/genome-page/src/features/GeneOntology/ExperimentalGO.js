import React, { Component } from 'react'

export default class ExperimentalGO extends Component {
    render() {
        return (
            <div>
                Experimental
            </div>
        )
    }
}