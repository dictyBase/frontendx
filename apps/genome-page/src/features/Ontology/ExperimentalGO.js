// @flow
import React from "react"
import { connect } from "react-redux"
import Grid from "@material-ui/core/Grid"
import WithGoDataRendering from "./WithGoDataRendering"
import PanelWrapper from "common/components/PanelWrapper"

type Props = {
  /** The data fetched from the API */
  goa: Object,
}

/**
 * Filters and shares only the Experimental GO data (IMP, IGI, IDA)
 */

export const ExperimentalGO = (props: Props) => {
  // const data = props.goa.data.filter(
  //   code =>
  //     code.evidence === "IMP" ||
  //     code.evidence === "IGI" ||
  //     code.evidence === "IDA",
  // )

  return (
    <Grid container>
      <Grid item sm={12} md={12} lg={12} xl={12}>
        <PanelWrapper title="Molecular Function">
          <div>data table coming later</div>
        </PanelWrapper>
        <PanelWrapper title="Biological Process">
          <div>data table coming later</div>
        </PanelWrapper>
        <PanelWrapper title="Cellular Composition">
          <div>data table coming later</div>
        </PanelWrapper>
      </Grid>
    </Grid>
  )
}

const mapStateToProps = ({ goa }) => ({ goa })

// $FlowFixMe
export default connect(
  mapStateToProps,
  null,
)(WithGoDataRendering(ExperimentalGO))
