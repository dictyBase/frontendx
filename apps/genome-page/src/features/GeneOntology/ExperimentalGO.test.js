import React from "react"
import renderer from "react-test-renderer"
import { ExperimentalGO } from "./ExperimentalGO"
jest.mock("react-virtualized")

test("matching a snapshot of ExperimentalGO", () => {
  const data = [
    {
      term: "ATP-gated ion channel activity",
      evidence: "IMP",
      with: "",
      reference: "Sivaramakrishnan & Fountain (2012)",
      date: "21-02-2014",
      source: "DDB",
    },
    {
      term: "calcium ion transmembrane transport requires ATP",
      evidence: "IGI",
      with: "p2xA p2xC p2xD p2xE",
      reference: "Sivaramakrishnan & Fountain (2012)",
      date: "21-02-2014",
      source: "DDB",
    },
  ]
  const fetchData = () => {}

  const component = renderer.create(
    <ExperimentalGO data={data} fetchData={fetchData} />,
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
