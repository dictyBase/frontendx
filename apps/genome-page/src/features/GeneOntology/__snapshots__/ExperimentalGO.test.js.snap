// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matching a snapshot of ExperimentalGO 1`] = `
<div>
  <div
    className="sc-eilVRo bGduNt"
  >
    <h3>
      Molecular Function
    </h3>
  </div>
  <div
    className="wrapper"
  />
  <br />
  <div
    className="sc-eilVRo bGduNt"
  >
    <h3>
      Biological Process
    </h3>
  </div>
  <div
    className="wrapper"
  />
  <br />
  <div
    className="sc-eilVRo bGduNt"
  >
    <h3>
      Cellular Composition
    </h3>
  </div>
  <div
    className="wrapper"
  />
</div>
`;
